corehq/apps/commtrack/sms.py

```diff
@@ -318,6 +321,40 @@ def looks_like_prod_code(self, code):
         except ValueError:
             return True
 
+    def parse(self, text):
```

Comment:
There's a bit that could be extracted. Once Ro confirms this is the correct understanding of requirements I'll do that.